exporter/sumologicexporter/sender_test.go

```diff
@@ -0,0 +1,255 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumologicexporter
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+)
+
+type senderTest struct {
+	srv *httptest.Server
+	exp component.LogsExporter
+	s   *sender
+}
+
+func prepareSenderTest(t *testing.T, cb []func(req *http.Request)) *senderTest {
+	reqCounter := 0
+	// generate a test server so we can capture and inspect the request
+	testServer := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		res.WriteHeader(200)
+		res.Write([]byte(""))
+		if len(cb) > 0 && assert.Greater(t, len(cb), reqCounter) {
+			cb[reqCounter](req)
+			reqCounter++
+		}
+	}))
+
+	cfg := &Config{
+		URL:                testServer.URL,
+		LogFormat:          "text",
+		Client:             "otelcol",
+		MaxRequestBodySize: 20_971_520,
+		TimeoutSettings:    CreateDefaultTimeoutSettings(),
+	}
+	factory := NewFactory()
+	exp, err := factory.CreateLogsExporter(context.Background(), component.ExporterCreateParams{Logger: zap.NewNop()}, cfg)
+	assert.NoError(t, err)
```

Comment:
```suggestion
	require.NoError(t, err)
```

ditto below, no need to continue the test